### DISCUSSION TITLE
Visual Studio extension not working as good as in VSC

### DISCUSSION THREAD

**Discussion by leoshusar** [community/community#49427] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I have already commented under #48470, but my problem was quite different, so I'm moving it here into new thread.

I have a problem with Copilot in VS. Sometimes it works fine, but sometimes it doesn't or works really dumb.
Recently this happened (inside the comment is what Copilot suggested):
```cs
var newPoints = CalculatePoints(_points);
var newInnerPoints = //_innerPoints;
```
So I tried to help it:
```cs
var newPoints = CalculatePoints(_points);
var newInnerPoints = CalculatePoints(//CalculatePoints(_innerPoints);
```

Here is a video of how it doesn't suggest until I finish a word or trigger it manually:

https://user-images.githubusercontent.com/13891066/221575069-a4b32135-bd99-4fc6-9238-63a5c75548a2.mp4

Here is the dumbness where it suggests nonsense:

![image](https://user-images.githubusercontent.com/13891066/221577534-8905ee96-c6dc-4557-aa29-b2ef75d33998.png)

I deleted `_usedPolygons.ToList()` from the `ConvertPolygonsToObjectToCut()` method, that's why it's underlined, Copilot suggested the exact method body before I deleted it.

I tried to help it again:
![image](https://user-images.githubusercontent.com/13891066/221578315-81ef902c-20ac-41f4-a88a-1637edecbd12.png)

Here VSC:
![image](https://user-images.githubusercontent.com/13891066/221578425-46968d2c-22df-42cb-98ba-d8c368e0a3be.png)

And this happened just now:

![image](https://user-images.githubusercontent.com/13891066/223478969-2701dd3d-0a91-46f0-b270-a9e3b3dca5a8.png)

Why would it even suggest this?
Again, here VSC got it on the first try:
![image](https://user-images.githubusercontent.com/13891066/223479287-b950195a-df1f-41ed-9cd7-5a72ce9a0f33.png)

Overall I have much better experience in VSC with Copilot. There it gives better suggestions and also much faster. As you can see, in VS it sometimes doesn't give anything. Sometimes even manual trigger doesn't work in VS.

This is happening in both VS 17.5 stable and preview (two separate devices), Copilot extension on 1.77, but this happens for me since older versions.
![image](https://user-images.githubusercontent.com/13891066/223480033-5a6fa321-ae54-4fb7-90c1-98eeac944d55.png)

![image](https://user-images.githubusercontent.com/13891066/223480975-99904381-c745-4279-afc8-1e21201e3382.png)

![image](https://user-images.githubusercontent.com/13891066/223481012-d35607f8-4fa4-40d8-af33-3537c8cdf9e6.png)

EDIT: another one
![image](https://user-images.githubusercontent.com/13891066/223510140-59951b15-12c6-4c82-b4e7-1f4ba1a2c765.png)

![image](https://user-images.githubusercontent.com/13891066/223539589-dc9ebd5c-0fde-443a-ae69-570c258b95ec.png)

**Comment by KenBayer** [comment]
I have been experiencing similar issues with the visual studio extension for the last couple weeks. VS Code provides great suggestions, but the Visual Studio suggestions rarely work at all.

This is on VS Community 17.5.1 and extension version 1.77.

I just tried uninstalling and reverting back to extension version 1.71, and everything works great now. I get the same suggestions in both VS and VS Code. So it seems that there was a significant regression sometime after 1.71.
